### DISCUSSION TITLE
nanodet model does not has CPU support for inference

### DISCUSSION THREAD

**Discussion by pyther-hub** [openvinotoolkit/openvino_notebooks#907] [Q&A]
For GSoC 2023, I have chosen the [Nanodet model](https://github.com/RangiLyu/nanodet), which currently supports predictions only for GPU devices and not for CPU. I have identified an [issue](https://github.com/RangiLyu/nanodet/issues/18) related to this limitation, which is indicated in the [code](https://github.com/RangiLyu/nanodet/blob/dcc7e798adbb9e64c406173a0792b934dbd76497/nanodet/model/arch/one_stage_detector.py#L48) through the use of ".cuda()".

**Comment by blaz-r** [comment]
I'm not entirely sure, but I think it should also work on the CPU. I think that the issue you linked only specifies that training can't be done on the CPU, but in the case of the notebook you want inference, so I think it should work. The repository also includes [OpenVINO section](https://github.com/RangiLyu/nanodet/blob/main/demo_openvino/README.md) which also has the option to export to ONNX, so maybe look into that.
The issue also says something about removing `torch.cuda.synchronize()` sections in order for it to work on the CPU as well.
Hope that helps :)

**Reply by pyther-hub** [reply]
but if I remove that section then I have to change the source code also

**Reply by blaz-r** [reply]
In some cases you can change only this particular function by overriding it on already instantiated model.
Did you also check OpenVINO section for conversion procedure?

**Reply by pyther-hub** [reply]
Yes sir I saw that section I am able to convert the model. for the time being I have edited the source code so that it works for cpu

**Comment by pyther-hub** [comment]
to be precise if we look [here](https://github.com/RangiLyu/nanodet/blob/dcc7e798adbb9e64c406173a0792b934dbd76497/nanodet/model/arch/one_stage_detector.py#L50) there we have cuda.sun I need to remove these lines so this works properly
